main.tf

```diff
@@ -1,13 +1,204 @@
-resource "random_integer" "example" {
-  count = module.this.enabled ? 1 : 0
+locals {
+  enabled                 = module.this.enabled
+  create_log_bucket       = var.bucket_id == null ? true : false
+  bucket_id               = var.bucket_id != "" ? var.bucket_id : module.ssm_patch_log_s3_bucket.bucket_id
+  default_allowed_actions = ["s3:GetObject", "s3:PutObject", "s3:PutObjectAcl", "s3:GetEncryptionConfiguration"]
+}
+
+module "ssm_patch_log_s3_bucket" {
+  count                  = local.create_log_bucket ? 1 : 0
+  source                 = "cloudposse/s3-bucket/aws"
+  version                = "0.38.0"
+  acl                    = "private"
+  versioning_enabled     = false
+  allowed_bucket_actions = var.ssm_bucket_policy != null ? [] : local.default_allowed_actions
```

Comment:
why we mix `policy` and `action` here?
maybe `var. ssm_bucket_policy` needs to be renamed to `var. ssm_bucket_actions`?

Reply:
tha idea was to give flexibility to specify a custom policy otherwise the action will be taken in place.

Reply:
@jamengual 
if you look here https://github.com/cloudposse/terraform-aws-s3-bucket/blob/master/main.tf#L229
it does not work like that.
The caller of the module can provide a bucket policy (it goes to the bucket itself).
But `allowed_bucket_actions` is not about the bucket, it get attached to the IAM user if the user is enabled

If you want to have what you wanted 
```
the idea was to give flexibility to specify a custom policy otherwise the action will be taken in place
```

If the caller does not provide `var.ssm_bucket_policy`, you need to create a bucket policy in this module using the actions in `default_allowed_actions`,  and give the bucket policy to the bucket 

```
policy                 = var.ssm_bucket_policy != null && var.ssm_bucket_policy != "" ? var.ssm_bucket_policy : <created_bucket_policy>
``

Reply:
understood